### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by jantimon** [pixelass/one-listener#4]
Right now `one-listner` just redirects all events to request animation frame.

For performance reasons you might also add a throttle.

**Comment by pixelass** [comment]
the idea is to add an optional throttle and a limit

if the limit is reached the throttle will kick in,

Some ideas:

```js
requestEventListener('scroll', ()=>{/*do something*/}, 300) // forced throttle
```
... would throttle only this function but always

```js
oneListener.throttle(300)// throttles 300ms (default: 100)
oneListener.limit(20)// throttles if more than 20 handlers are set (default: 10)
requestEventListener('scroll', ()=>{/*do something*/})
```
... will throttle the eventListener itself therefore all handlers have the same throttle 

Open to hear ideas

**Comment by pixelass** [comment]
I added a simple implementation here: https://github.com/pixelass/one-listener/tree/feature/limit-throttle

This works with a global store which could be modified from anywhere therefore it might be better to allow creating instances. 

What this does:

* check the limit and apply the settings
* each event has to reach its own limit (throttled scrolls != throttled resizes)

I still want the ability to create one global instance and then restrict the `limit` and `throttle` setters.


e.g.  wrap your own module with settings applied and then export for further usage

```js 
/**
 * create one listener for you app and then export it
 */
import OneListener from 'one-listener'
const one = new OneListener();
one.limit(10)
one.throttle(40)
export.requestEventListener = one.requestEventListener
export.requestEventListener = one.cancelEventListener
```

**Comment by pixelass** [comment]
check this one: https://github.com/pixelass/one-listener/tree/feature/constructor

This allows to create own locked modules 

```js
import OneListener from 'one-listener';

// create locked instances
const one = new OneListener({
    limit: 6,
    throttle: 200
});
const never = new OneListener({
    limit: Infinity
});
const always = new OneListener({
    limit: 0,
    throttle: 100
});

const listeners = {
  one: {one.requestEventListener,one.cancelEventListener},
  always: {always.requestEventListener,always.cancelEventListener},
  never: {requestEventListener,cancelEventListener}
}

// export locked instances
export default listeners

// then import

import listeners from './listeners';
const {one,never,always}
const {requestEventListener, cancelEventListener} = one;

//requestEventListener() || one.requestEventListener()
//always.requestEventListener()
//never.requestEventListener()

```

**Comment by pixelass** [comment]
fixed by #6
